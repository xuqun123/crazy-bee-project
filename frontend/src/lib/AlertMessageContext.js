import { createContext } from 'react'

const AlertMessageContext = createContext({
  alert: { message: '', severity: 'success' },
  setAlert: () => {},
})
AlertMessageContext.displayName = 'AlertMessageContext'

export default AlertMessageContext
