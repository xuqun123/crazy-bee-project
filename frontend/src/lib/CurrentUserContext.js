import { createContext } from 'react'

const CurrentUserContext = createContext(null)
CurrentUserContext.displayName = 'CurrentUserContext'

export default CurrentUserContext
