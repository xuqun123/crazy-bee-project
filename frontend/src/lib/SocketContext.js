import { createContext } from 'react'

const SocketContext = createContext(null)
SocketContext.displayName = 'SocketContext'

export default SocketContext
