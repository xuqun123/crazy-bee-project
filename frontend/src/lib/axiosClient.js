import axios from 'axios'

const axiosClient = axios.create({
  baseURL: process.env.REACT_APP_API_BASE_URL,
  headers: {
    'Content-Type': 'application/json',
    Authorization: localStorage.getItem('jwt') ? `Bearer ${localStorage.getItem('jwt')}` : '',
  },
})

export default axiosClient
