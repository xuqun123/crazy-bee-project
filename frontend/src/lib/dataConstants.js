const defaultNFTCollectionsLimit = 7

export { defaultNFTCollectionsLimit }
