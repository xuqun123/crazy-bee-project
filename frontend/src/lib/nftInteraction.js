const installMetamaskError = {
  address: '',
  message: (
    <span>
      ðŸ¦Š{' '}
      <a
        style={{ color: '#fff' }}
        target="_blank"
        href={`https://metamask.io/download`}
        rel="noreferrer"
      >
        To connect your wallet, <strong>you must install Metamask extenstion</strong> in your
        browser first!
      </a>
    </span>
  ),
  severity: 'warning',
}

const connectToWallet = async () => {
  if (window.ethereum) {
    try {
      const addressArray = await window.ethereum.request({
        method: 'eth_requestAccounts',
      })
      const obj = {
        message: 'Your wallet has been connected successfully!',
        address: addressArray[0],
      }
      return obj
    } catch (err) {
      return {
        address: '',
        message: 'ðŸ˜¥ ' + err.message,
        severity: 'error',
      }
    }
  } else {
    return installMetamaskError
  }
}

const getConnectedWallet = async () => {
  if (window.ethereum) {
    try {
      const addressArray = await window.ethereum.request({
        method: 'eth_accounts',
      })
      if (addressArray.length > 0) {
        return {
          address: addressArray[0],
          message: 'Your wallet has been connected successfully!',
        }
      } else {
        return {
          address: '',
          message: 'ðŸ¦Š Please connect to Metamask wallet using the top right button!',
        }
      }
    } catch (err) {
      return {
        address: '',
        message: 'ðŸ˜¥ Wallet connection failed: ' + err.message,
        severity: 'error',
      }
    }
  } else {
    return installMetamaskError
  }
}

export { connectToWallet, getConnectedWallet }
