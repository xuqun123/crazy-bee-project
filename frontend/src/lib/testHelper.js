const fakeUser = {
  _id: '63c551e0ca77037c5a8a017f',
  username: 'abc123',
  firstName: 'Luella',
  lastName: 'Glover',
  walletAddresses: ['57541aa3-11d7-4398-a5db-c3ac2eed517e'],
  avatarUrl:
    'https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/297.jpg',
  bannerImageUrl: 'https://loremflickr.com/1024/200/nature',
  bio: 'Aliquid iste soluta harum. Modi nobis sequi laudantium dolorem dolorum consectetur omnis ipsam. Perspiciatis deleniti atque temporibus labore soluta deleniti quisquam. Eos itaque consectetur similique beatae maiores ipsum eveniet. Dolore vitae fugit. Sint perferendis amet vero. Dolore vero sit magnam unde mollitia cum labore. Beatae accusantium dolorem veniam nulla ex earum tempore. Ea quaerat perferendis officiis ipsa molestiae rem hic consequatur.',
  dob: '2015-10-21T10:25:06.923Z',
  createdAt: '2023-01-16T13:32:16.716Z',
  updatedAt: '2023-01-20T11:15:36.637Z',
}

const fakeNftCollection = {
  name: 'a testing nft collection',
  _id: '1234-5678-abcd',
  summary: 'this is a testing nft collection',
}

export { fakeUser, fakeNftCollection }
