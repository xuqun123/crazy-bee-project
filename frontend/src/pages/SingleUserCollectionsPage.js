import { useState, useEffect } from 'react'
import { useParams } from 'react-router-dom'
import NFTCollectionsList from '../components/NFTCollectionsList'
import axiosClient from '../lib/axiosClient'
import UserSummary from '../components/UserSummary'

function SingleUserCollectionsPage() {
  const { userId } = useParams()
  const [user, setUser] = useState(null)
  const [loading, setLoading] = useState(true)

  useEffect(() => {
    axiosClient
      .get(`/users/${userId}`)
      .then((response) => {
        setLoading(false)
        setUser(response?.data?.data)
      })
      .catch((error) => {
        setLoading(false)
        const message = `Get nftCollections data failed: ${error.message}`

        console.error(message)
      })
  }, [userId])

  return (
    <>
      <UserSummary userId={userId} user={user} loading={loading} enableCreate={true} />
      <NFTCollectionsList
        userId={userId}
        enableLoadMore={true}
        enableSearch={true}
        enableEdit={true}
        enableDelete={true}
      />
    </>
  )
}

export default SingleUserCollectionsPage
