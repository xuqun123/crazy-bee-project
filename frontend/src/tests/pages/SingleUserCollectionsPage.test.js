import React from 'react'
import { render, screen, waitFor, fireEvent } from '@testing-library/react'
import SingleUserCollectionsPage from '../../pages/SingleUserCollectionsPage'
import axiosClient from '../../lib/axiosClient'
import { fakeUser, fakeNftCollection } from '../../lib/testHelper'
import { MemoryRouter } from 'react-router-dom'

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => {
    const { fakeUser } = require('../../lib/testHelper')

    return {
      userId: fakeUser._id,
    }
  },
}))

describe('SingleUserCollectionsPage', () => {
  afterEach(() => jest.clearAllMocks())

  describe('get users data via api', () => {
    beforeEach(() => {
      jest.spyOn(axiosClient, 'get').mockImplementation((url) => {
        if (url.includes('users')) {
          return Promise.resolve({ data: { data: { ...fakeUser } } })
        } else if (url.includes('nftCollections')) {
          return Promise.resolve({ data: { data: [fakeNftCollection], loadMore: true } })
        }
      })
    })

    it('render the SingleUserCollectionsPage view properly with expected elements', async () => {
      const { rerender } = render(<SingleUserCollectionsPage />, { wrapper: MemoryRouter })
      let usernameHeader

      await waitFor(() => {
        usernameHeader = screen.getByText(`@${fakeUser.username}`)
      })
      expect(usernameHeader).toBeInTheDocument()

      const userBioSection = screen.getByText(fakeUser.bio)
      expect(userBioSection).toBeInTheDocument()

      const loadMoreButton = screen.getByTestId('load-more-btn')
      expect(loadMoreButton).toBeInTheDocument()
      rerender(<SingleUserCollectionsPage />, { wrapper: MemoryRouter })

      fireEvent.click(loadMoreButton)
      const userBioSections = await screen.findByText(fakeUser.bio)
      expect(userBioSections).toBeInTheDocument()
    })

    it('snapshot the SingleUserCollectionsPage view with all expected elements', async () => {
      const view = render(<SingleUserCollectionsPage />, { wrapper: MemoryRouter })

      await waitFor(() => {
        expect(screen.getByText(`@${fakeUser.username}`)).toBeInTheDocument()
      })
      expect(view).toMatchSnapshot()
    })
  })

  describe('failed to get the user data', () => {
    const error = new Error('something goes wrong')
    let mockConsoleError

    beforeEach(() => {
      jest.spyOn(axiosClient, 'get').mockRejectedValue(error)
      mockConsoleError = jest.spyOn(console, 'error').mockImplementation((error) => error)
    })

    it('render the SingleUserCollectionsPage view without rendering any user data', async () => {
      render(<SingleUserCollectionsPage />, { wrapper: MemoryRouter })

      await waitFor(() => {
        expect(screen.queryByText(fakeUser.username)).not.toBeInTheDocument()
      })

      expect(mockConsoleError).toHaveBeenCalled()
    })
  })
})
