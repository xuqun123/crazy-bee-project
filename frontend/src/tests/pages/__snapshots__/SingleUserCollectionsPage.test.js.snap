// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleUserCollectionsPage get users data via api snapshot the SingleUserCollectionsPage view with all expected elements 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt="banner"
        height="300px"
        src="https://loremflickr.com/1024/200/nature"
        width="100%"
      />
      <div
        class="MuiGrid-root MuiGrid-container css-3tr9xy-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1het00i-MuiGrid-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-1cjwto2-MuiAvatar-root"
          >
            <img
              alt="avatar-abc123"
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/297.jpg"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 css-9drkqd-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-gg4vpm"
            direction="row"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 MuiTypography-alignLeft css-fokda-MuiTypography-root"
            >
              @
              abc123
            </h4>
            <a
              href="/users/63c551e0ca77037c5a8a017f/collections/new"
              style="text-decoration: none;"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium css-ke5b6m-MuiButtonBase-root-MuiButton-root"
                data-testid="create-nftcollection-btn"
                tabindex="0"
                type="button"
              >
                Create
              </button>
            </a>
          </div>
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-alignLeft css-1fh2sue-MuiTypography-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1hd5cgq-MuiSvgIcon-root"
              data-testid="TokenIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.97 6.43 12 2 4.03 6.43 9.1 9.24C9.83 8.48 10.86 8 12 8s2.17.48 2.9 1.24l5.07-2.81zM10 12c0-1.1.9-2 2-2s2 .9 2 2-.9 2-2 2-2-.9-2-2zm1 9.44L3 17V8.14l5.13 2.85c-.09.32-.13.66-.13 1.01 0 1.86 1.27 3.43 3 3.87v5.57zm2 0v-5.57c1.73-.44 3-2.01 3-3.87 0-.35-.04-.69-.13-1.01L21 8.14V17l-8 4.44z"
              />
            </svg>
            0x57541aa3-11d7-4398-a5db-c3ac2eed517e
             
          </h5>
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-alignLeft css-i32owz-MuiTypography-root"
          >
            <span
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignLeft css-11uw5h0-MuiTypography-root"
            >
              Joined Since
               
            </span>
            17 Jan 2023
          </h5>
          <br />
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft css-11qvwo8-MuiTypography-root"
          >
            Aliquid iste soluta harum. Modi nobis sequi laudantium dolorem dolorum consectetur omnis ipsam. Perspiciatis deleniti atque temporibus labore soluta deleniti quisquam. Eos itaque consectetur similique beatae maiores ipsum eveniet. Dolore vitae fugit. Sint perferendis amet vero. Dolore vero sit magnam unde mollitia cum labore. Beatae accusantium dolorem veniam nulla ex earum tempore. Ea quaerat perferendis officiis ipsa molestiae rem hic consequatur.
          </h6>
        </div>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1mknzg7-MuiContainer-root"
        width="lg"
      >
        <form
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-paob4f-MuiPaper-root"
        >
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary css-piqts5-MuiInputBase-root"
          >
            <input
              aria-label="Search NFT collections"
              class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
              placeholder="Search NFT collections"
              type="text"
              value=""
            />
          </div>
          <button
            aria-label="search"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1gtpq4r-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical css-jbezd2-MuiDivider-root"
          />
          <button
            aria-label="directions"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1bxg8jb-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="RestartAltIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 5V2L8 6l4 4V7c3.31 0 6 2.69 6 6 0 2.97-2.17 5.43-5 5.91v2.02c3.95-.49 7-3.85 7-7.93 0-4.42-3.58-8-8-8zm-6 8c0-1.65.67-3.15 1.76-4.24L6.34 7.34C4.9 8.79 4 10.79 4 13c0 4.08 3.05 7.44 7 7.93v-2.02c-2.83-.48-5-2.94-5-5.91z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </form>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 css-6w982j-MuiGrid-root"
          >
            <a
              href="/collections/63cce9886f124b9e51502125"
              style="text-decoration: none;"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-3mgkv0-MuiPaper-root-MuiCard-root"
              >
                <img
                  alt="cover-image"
                  class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                  src="https://loremflickr.com/640/480/animals"
                />
                <div
                  class="MuiCardContent-root css-2r3irp-MuiCardContent-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-p css-37jdci-MuiTypography-root"
                  >
                    1st December 2022
                  </p>
                  <h2
                    class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
                  >
                    dolor alias adipisci vel cumque labore
                  </h2>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    Natus praesentium inventore ab magni eveniet earum.
                  </p>
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1xur0t1-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      document
                    </span>
                  </div>
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1o92u0b-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      video
                    </span>
                  </div>
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1yl503k-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      audio
                    </span>
                  </div>
                </div>
                <div
                  class="MuiCardActions-root MuiCardActions-spacing css-15f5lf7-MuiCardActions-root"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeSmall MuiButton-containedSizeSmall css-1uaqrku-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Edit
                  </button>
                </div>
              </div>
            </a>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1njilb0-MuiButtonBase-root-MuiButton-root"
          data-testid="load-more-btn"
          tabindex="0"
          type="button"
        >
          Load More
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <img
      alt="banner"
      height="300px"
      src="https://loremflickr.com/1024/200/nature"
      width="100%"
    />
    <div
      class="MuiGrid-root MuiGrid-container css-3tr9xy-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1het00i-MuiGrid-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-1cjwto2-MuiAvatar-root"
        >
          <img
            alt="avatar-abc123"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/297.jpg"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 css-9drkqd-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-gg4vpm"
          direction="row"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 MuiTypography-alignLeft css-fokda-MuiTypography-root"
          >
            @
            abc123
          </h4>
          <a
            href="/users/63c551e0ca77037c5a8a017f/collections/new"
            style="text-decoration: none;"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium css-ke5b6m-MuiButtonBase-root-MuiButton-root"
              data-testid="create-nftcollection-btn"
              tabindex="0"
              type="button"
            >
              Create
            </button>
          </a>
        </div>
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-alignLeft css-1fh2sue-MuiTypography-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1hd5cgq-MuiSvgIcon-root"
            data-testid="TokenIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19.97 6.43 12 2 4.03 6.43 9.1 9.24C9.83 8.48 10.86 8 12 8s2.17.48 2.9 1.24l5.07-2.81zM10 12c0-1.1.9-2 2-2s2 .9 2 2-.9 2-2 2-2-.9-2-2zm1 9.44L3 17V8.14l5.13 2.85c-.09.32-.13.66-.13 1.01 0 1.86 1.27 3.43 3 3.87v5.57zm2 0v-5.57c1.73-.44 3-2.01 3-3.87 0-.35-.04-.69-.13-1.01L21 8.14V17l-8 4.44z"
            />
          </svg>
          0x57541aa3-11d7-4398-a5db-c3ac2eed517e
           
        </h5>
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-alignLeft css-i32owz-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-h5 MuiTypography-alignLeft css-11uw5h0-MuiTypography-root"
          >
            Joined Since
             
          </span>
          17 Jan 2023
        </h5>
        <br />
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft css-11qvwo8-MuiTypography-root"
        >
          Aliquid iste soluta harum. Modi nobis sequi laudantium dolorem dolorum consectetur omnis ipsam. Perspiciatis deleniti atque temporibus labore soluta deleniti quisquam. Eos itaque consectetur similique beatae maiores ipsum eveniet. Dolore vitae fugit. Sint perferendis amet vero. Dolore vero sit magnam unde mollitia cum labore. Beatae accusantium dolorem veniam nulla ex earum tempore. Ea quaerat perferendis officiis ipsa molestiae rem hic consequatur.
        </h6>
      </div>
    </div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1mknzg7-MuiContainer-root"
      width="lg"
    >
      <form
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-paob4f-MuiPaper-root"
      >
        <div
          class="MuiInputBase-root MuiInputBase-colorPrimary css-piqts5-MuiInputBase-root"
        >
          <input
            aria-label="Search NFT collections"
            class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
            placeholder="Search NFT collections"
            type="text"
            value=""
          />
        </div>
        <button
          aria-label="search"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1gtpq4r-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical css-jbezd2-MuiDivider-root"
        />
        <button
          aria-label="directions"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1bxg8jb-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="RestartAltIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 5V2L8 6l4 4V7c3.31 0 6 2.69 6 6 0 2.97-2.17 5.43-5 5.91v2.02c3.95-.49 7-3.85 7-7.93 0-4.42-3.58-8-8-8zm-6 8c0-1.65.67-3.15 1.76-4.24L6.34 7.34C4.9 8.79 4 10.79 4 13c0 4.08 3.05 7.44 7 7.93v-2.02c-2.83-.48-5-2.94-5-5.91z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 css-6w982j-MuiGrid-root"
        >
          <a
            href="/collections/63cce9886f124b9e51502125"
            style="text-decoration: none;"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-3mgkv0-MuiPaper-root-MuiCard-root"
            >
              <img
                alt="cover-image"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                src="https://loremflickr.com/640/480/animals"
              />
              <div
                class="MuiCardContent-root css-2r3irp-MuiCardContent-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-p css-37jdci-MuiTypography-root"
                >
                  1st December 2022
                </p>
                <h2
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
                >
                  dolor alias adipisci vel cumque labore
                </h2>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  Natus praesentium inventore ab magni eveniet earum.
                </p>
                <div
                  class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1xur0t1-MuiChip-root"
                >
                  <span
                    class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                  >
                    document
                  </span>
                </div>
                <div
                  class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1o92u0b-MuiChip-root"
                >
                  <span
                    class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                  >
                    video
                  </span>
                </div>
                <div
                  class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1yl503k-MuiChip-root"
                >
                  <span
                    class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                  >
                    audio
                  </span>
                </div>
              </div>
              <div
                class="MuiCardActions-root MuiCardActions-spacing css-15f5lf7-MuiCardActions-root"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeSmall MuiButton-containedSizeSmall css-1uaqrku-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Edit
                </button>
              </div>
            </div>
          </a>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1njilb0-MuiButtonBase-root-MuiButton-root"
        data-testid="load-more-btn"
        tabindex="0"
        type="button"
      >
        Load More
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
